### PR TITLE
Declare few more files as `DocumentationChanges`

### PR DESCRIPTION
**Pull request by IT-VBFK**
Closes: #2098 

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

File: Build/Build.cs [hunk 1]
```diff
@@ -287,7 +287,12 @@ class Build : NukeBuild
         });
 
     bool HasDocumentationChanges =>
-        Changes.Any(x => x.StartsWith("docs"));
+        Changes.Any(x =>
```
Comment (dennisdoomen) [inline Build/Build.cs:290]: 🤔 Shouldn't we also trigger this if anything in `_build` folder changes?

Comment (IT-VBFK) [reply Build/Build.cs:290]: On documentation changes?

Comment (IT-VBFK) [reply Build/Build.cs:290]: Where is the `_build` folder btw?

File: Build/Build.cs [hunk 2]
```diff
@@ -287,7 +287,12 @@ class Build : NukeBuild
...
     bool HasDocumentationChanges =>
-        Changes.Any(x => x.StartsWith("docs"));
+        Changes.Any(x => 
+            x.StartsWith("docs") || 
+            x.StartsWith("CONTRIBUTING.md") ||
+            x.StartsWith("cSpell.json") ||
+            x.StartsWith("LICENSE") ||
+            x.StartsWith("README.md"));
 
     bool HasSourceChanges =>
```
Comment (jnyrup) [inline Build/Build.cs:297]: I think this needs to be structured something like this to ensure that `HasSourceChanges` is the complement of `HasDocumentationChanges`.

```cs
bool HasDocumentationChanges => Changes.Any(x => x.IsDocumentation());

bool HasSourceChanges=> Changes.Any(x => !x.IsDocumentation());

static bool IsDocumentation(string x)
{
    return x.StartsWith("docs") || 
           x.StartsWith("CONTRIBUTING.md") ||
           x.StartsWith("cSpell.json") ||
           x.StartsWith("LICENSE") ||
           x.StartsWith("README.md");
}
```

File: Build/Build.cs (hunk 1 above)
Comment (dennisdoomen) [reply Build/Build.cs:290]: The folder is called `Build`. The project is called `build.csproj`

If something changes related to how to do spell checks or we update any dependencies related to that, you also want to run the full build. Right?

File: Build/Build.cs (hunk 2 above)
Comment (IT-VBFK) [reply Build/Build.cs:297]: Yes, I wanted to refactor this, but focused on the "build clean up PR".

Now I can do that

**Review by jnyrup — APPROVED** [review]

File: Build/Build.cs (hunk 1 above)
Comment (IT-VBFK) [reply Build/Build.cs]: Sure.. I've forgotten that either :D

File: Build/Build.cs [hunk 3]
```diff
@@ -286,11 +286,16 @@ class Build : NukeBuild
...
-    bool HasSourceChanges =>
-        Changes.Any(x => !x.StartsWith("docs"));
+    bool HasSourceChanges => Changes.Any(x => !IsDocumentation(x));
+
+    static bool IsDocumentation(string x) =>
+        x.StartsWith("docs") ||
+        x.StartsWith("CONTRIBUTING.md") ||
+        x.StartsWith("cSpell.json") ||
+        x.StartsWith("LICENSE") ||
+        x.StartsWith("README.md");
```
Comment (jnyrup) [inline Build/Build.cs:298]: Gave the repo another look, and [`package.json`](https://github.com/fluentassertions/fluentassertions/blob/develop/package.json) should also be added here.

Comment (IT-VBFK) [reply Build/Build.cs:298]: ok

**Review by jnyrup — APPROVED** [review]

**Comment by IT-VBFK** [comment]
Just rebased :)

**Review by dennisdoomen — APPROVED** [review]

**Comment by dennisdoomen** [comment]
Thanks for this contribution.

**Pull request merged**